app/helpers/admin/payroll_run_helper.rb

```diff
@@ -0,0 +1,14 @@
+module Admin
+  module PayrollRunHelper
+    def next_payroll_file_to_cantium_due_date
+      next_payrollable_month = PayrollRun.this_month.exists? ? Date.today.next_month : Date.today
+      all_fridays_in_month(next_payrollable_month)[-3]
+    end
+
+    private
+
+    def all_fridays_in_month(date)
+      (date.beginning_of_month..date.end_of_month).select { |day| day.friday? }
```

Comment:
For this I was thinking about putting `next_payrollable_month` in a method, as you had before

```ruby
def next_payrollable_month
      target_date = PayrollRun.this_month.any? ? Date.today.next_month : Date.today
      (target_date.beginning_of_month..target_date.end_of_month)
end
```

That way, you end up returning a `Range`, which probably more accurately represents a whole month, then you can get all the fridays in the month like so:

```ruby
def all_fridays_in_month(month)
  month.select { |day| day.friday? }
end
```

Reply:
I do like it actually being a month that is passed, I'm now here:

````
  module PayrollRunHelper
    def next_payroll_file_to_cantium_due_date
      next_payrollable_month = PayrollRun.this_month.exists? ? Date.today.next_month.all_month : Date.today.all_month
      all_fridays_in_month(next_payrollable_month)[-3]
    end

    private

    def all_fridays_in_month(month)
      month.select { |day| day.friday? }
    end
  end
````

Reply:
Ah, yeah, that old fella `all_month`, good call 👍